### PR TITLE
COMP: Update CMake minimum required version from 3.5 to 3.13

### PR DESCRIPTION
**Pull request by phcerdan**
Discourse post:
https://discourse.slicer.org/t/updating-minimum-required-version-of-cmake-to-3-13/6012

// ----------
Update MultiVolumeExplorer:

```
git shortlog --no-merges 0bca789..16180487
Pablo Hernandez-Cerdan (1):
      Update cmake_minimum_required to 3.13
```
// ----------
Update MultiVolumeImporter:

```
git shortlog --no-merges   e1c1cd8..ce324adb
Pablo Hernandez-Cerdan (1):
      Update cmake_minimum_required to 3.13
```
// ----------
Update SimpleFilters

```
git shortlog --no-merges   dd1e8be5..8c27c74b
Pablo Hernandez-Cerdan (1):
      Update cmake_minimum_required to 3.13
```
// ----------
Update OtsuThresholdImageFilter

```
git shortlog --no-merges  cf39e506..c14d5b8e
Pablo Hernandez-Cerdan (1):
      Update cmake_minimum_required to 3.13
```
// ----------
Update DataStore

```
git shortlog --no-merges  8053b028..5fcf3b64
Pablo Hernandez-Cerdan (1):
      Update cmake_minimum_required to 3.13
```
// ----------
Update CompareVolumes

```
git shortlog --no-merges b2a9a0d9..34f1b3da
Pablo Hernandez-Cerdan (1):
      Update cmake_minimum_required to 3.13
```

// ----------
Update LandmarkRegistration

```
git shortlog --no-merges 837e5edf..6569f69e
Pablo Hernandez-Cerdan (1):
      Update cmake_minimum_required to 3.13
```

Co-authored-by: Hans Johnson <hans-johnson@uiowa.edu>

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,16 +1,12 @@
-cmake_minimum_required(VERSION 3.5)
+cmake_minimum_required(VERSION 3.13)
 
 #-----------------------------------------------------------------------------
 # Enable C++11
 #-----------------------------------------------------------------------------
 set(_msg "Setting C++ standard")
 message(STATUS "${_msg}")
 if(NOT DEFINED CMAKE_CXX_STANDARD)
-  if("${Slicer_VTK_VERSION_MAJOR}" STREQUAL "8" OR DEFINED Qt5_DIR)
```
Comment (jcfr) [inline CMakeLists.txt]: Let's keep this in for now and apply it when effectively removing support for Qt4/VTK7.

File: CMake/ExternalProjectDependency.cmake [hunk 2]
```diff
@@ -474,18 +474,13 @@ function(_sb_get_external_project_arguments proj varname)
   list(APPEND _ep_arguments LIST_SEPARATOR ${EP_LIST_SEPARATOR})
 
   list(APPEND _ep_arguments CMAKE_GENERATOR ${_sb_CMAKE_GENERATOR})
```
Comment (jcfr) [inline CMake/ExternalProjectDependency.cmake]: Since this file is copied from https://github.com/commontk/Artichoke let's keep the file unmodified.

**Comment by phcerdan** [comment]
@hjmjohnson no please, the goal is to tackle the upgrade as a block, this is just a start, do not abandon any of your PR's yet, they will be merged (maybe to another branch, instead of master) and then closed.

**Comment by hjmjohnson** [comment]
@phcerdan Thanks.  Shall I rebase those PR's on the current master so that they are closer to up-to-date?

If there are mechanical things I can do, let me know.   PS:  I've also added you as a collaborator (and once accepted, I'll give you write access) to my slicer repo that has these patches.

 https://github.com/hjmjohnson/Slicer/invitations

That way if there are minor changes, that need to be made you can do that directly in the hjmjohnson repo to accelerate acceptance.

**Comment by phcerdan** [comment]
Thanks @hjmjohnson !

**Comment by jcfr** [comment]
Now  that the remove extensions (MultiVolumeExplorer, ... ) have been updated, consider updating the Git hash.

Commit message should also be updated to `COMP: Update CMake minimum required version from 3.5 to 3.13`. Look at [r26201](http://viewvc.slicer.org/viewvc.cgi/Slicer4?view=revision&revision=26201) and consider referencing the discourse post as well as any other relevant information in the commit message. 

Adding `Co-authored-by: Hans Johnson <hans-johnson@uiowa.edu>` would also be sensible.

**Comment by phcerdan** [comment]
Updated git hash to those remote extensions where PR's where merged.

PR opened but not merged yet:
- LandmarkRegistration: https://github.com/pieper/LandmarkRegistration/pull/26
- CompareVolumes: https://github.com/pieper/CompareVolumes/pull/10

No PR opened for:
- EMSegment
- BRAINSTools

**Comment by phcerdan** [comment]
Hey @jc,
how do I convince circleci here to use the [updated dockcross images](https://github.com/dockbuild/dockbuild/pull/56) that we just updated?

**Comment by jcfr** [comment]
>  how do I convince circleci here to use the updated dockcross images that we just updated?

Short answer: It will happen automatically

Long answer: This will happen automatically when the [slicer/slicer-base](https://github.com/thewtex/SlicerDocker#images) image is automatically updated [during the nightly build](https://github.com/Slicer/DashboardScripts/blob/bae66c5188eb4d482e4adadb551ce24405d1e7ee/metroplex.sh#L52-L54) which is itself using the [SlicerDockerUpdate](https://github.com/Slicer/SlicerDockerUpdate) scripts.

**Comment by jcfr** [comment]
After testing locally, I will move forward with the integration. That way, this will allow the next round of night build to validate the changes.

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.5)
+cmake_minimum_required(VERSION 3.13.0)
```
Comment (jcfr) [inline CMakeLists.txt]: Thinking more about this, consider setting `3.13.4`. It fixes few regressions and is the version tested.
See https://cmake.org/cmake/help/v3.13/release/3.13.html#id1

Comment (jcfr) [reply CMakeLists.txt]: Update: I will take care of the change when integrating.

Comment (phcerdan) [reply CMakeLists.txt]: I can open PR's for the remotes and update the hashes tomorrow

**Comment by jcfr** [comment]
Closing. Change integrated in [r27988](http://viewvc.slicer.org/viewvc.cgi/Slicer4?view=revision&revision=27988)

**Comment by jcfr** [comment]
> >    how do I convince circleci here to use the updated dockcross images that we just updated?

> Short answer: It will happen automatically

While the dockbuild images are rebuild automatically when PR are merged. I was incorrect regarding the slicer images. The `slicer/buildenv-qt5-centos7:latest` image has to be rebuilt explicitly and pushed to dockerhub. This is currently happening on metroplex build machine.

**Pull request closed without merging**